packages/teleterm/src/mainProcess/windowsManager.ts

```diff
@@ -0,0 +1,96 @@
+import { BrowserWindow, Rectangle, screen } from 'electron';
+import { getAssetPath } from 'teleterm/mainProcess/runtimeSettings';
+import path from 'path';
+import { FileStorage } from 'teleterm/services/fileStorage';
+import { RuntimeSettings } from 'teleterm/mainProcess/types';
+import theme from 'teleterm/ui/ThemeProvider/theme';
+
+type WindowState = Rectangle;
+
+export class WindowsManager {
+  private storageKey = 'windowState';
+
+  constructor(
+    private fileStorage: FileStorage,
+    private settings: RuntimeSettings
+  ) {}
+
+  createWindow(): void {
+    const windowState = this.getWindowState();
+    const window = new BrowserWindow({
+      x: windowState.x,
+      y: windowState.y,
+      width: windowState.width,
+      height: windowState.height,
+      backgroundColor: theme.colors.primary.main,
+      minWidth: 400,
+      minHeight: 300,
+      title: 'Teleport Terminal',
+      icon: getAssetPath('icon.png'),
+      webPreferences: {
+        contextIsolation: true,
+        nodeIntegration: false,
+        preload: path.join(__dirname, 'preload.js'),
+      },
+    });
+
+    window.once('close', () => {
+      this.saveWindowState(window);
+    });
+
+    if (this.settings.dev) {
+      window.loadURL('https://localhost:8080');
+    } else {
+      window.loadFile(path.join(__dirname, '../renderer/index.html'));
+    }
+  }
+
+  private saveWindowState(window: BrowserWindow): void {
+    const windowState: WindowState = {
+      ...window.getNormalBounds(),
```

Comment:
> Note: whatever the current state of the window : maximized, minimized or in fullscreen, this function always returns the position and size of the window in normal state. In normal state, getBounds and getNormalBounds returns the same [Rectangle](https://www.electronjs.org/docs/latest/api/structures/rectangle).

Did you experiment with using `getBounds` here? I don't have time to check it properly, but when I used `getBounds` here, the app was able to restore the window to the correct display when it was in a simple fullscreen mode and in normal fullscreen mode. In normal fullscreen mode it restored the app to just simple fullscreen, but I think that's okay for now.

Reply:
TBH, `getBounds` solves one problem (window is restored maximised), but it creates another (if you try to unmaxmize the window by using double click it became very small - in our case 400x300 - the minimal size). For me, the most correct way is to save dimensions from `normalBounds`, because to create other states (`fullScreen/maxmized`) we don't need any previous information.

I've just spend some time experimenting with `getBounds()`. I wanted to save both info, then use `getBounds()` to restore correct screen, and `getNormalBound()` to restore correct size and placement, but it still doesn't work for fullscreen mode :(